### PR TITLE
feat: adding a darker modal backdrop

### PR DESCRIPTION
**Pull request by anandtiwary**

### CODE REVIEW

File: projects/components/src/popover/popover.service.ts [hunk 1]
```diff
@@ -25,7 +25,7 @@ export class PopoverService {
     const overlayRef = this.overlay.create({
       positionStrategy: initialPositionStrategy,
       hasBackdrop: options.backdrop,
-      backdropClass: 'cdk-overlay-transparent-backdrop'
+      backdropClass: options.backdrop ? 'modal-overlay-backdrop' : 'cdk-overlay-transparent-backdrop'
```
Comment (aaron-steinfeld) [inline projects/components/src/popover/popover.service.ts:28]: This turns all backdrops dark. If I recall, backdrop class is unused if hasBackdrop is false, so the transparent class would never be used here. We would need to migrate backdrop the config boolean to an enum to the effect of [TRANSPARENT, OPAQUE, NONE] (or use undefined to indicate none)

Comment (anandtiwary) [reply projects/components/src/popover/popover.service.ts:28]: I didn't see any visual difference between None and transparent. Is there one?

Comment (aaron-steinfeld) [reply projects/components/src/popover/popover.service.ts:28]: Visually no, it's a behavior difference - with a transparent backdrop, interaction is blocked with anything outside the popover (and generally we'd close on a click on the backdrop). At least, that was the initial implementation as I recall.

Comment (anandtiwary) [reply projects/components/src/popover/popover.service.ts:28]: oh I see. Makes sense. Will add the new enum state

File: projects/components/src/popover/popover.service.ts [hunk 2]
```diff
@@ -78,4 +78,20 @@ export class PopoverService {
...
   }
+
+  private hasBackdrop(backdrop?: PopoverBackdrop): boolean {
+    return backdrop === PopoverBackdrop.Opaque || backdrop === PopoverBackdrop.Transparent;
+  }
+
+  private getBackdropClass(backdrop?: PopoverBackdrop): string {
+    switch (backdrop) {
+      case PopoverBackdrop.Transparent:
+        return 'cdk-overlay-transparent-backdrop';
```
Comment (aaron-steinfeld) [inline projects/components/src/popover/popover.service.ts:89]: not sure what's in this, but does it make sense to migrate to our own class for consistency? how much can a transparent backdrop have in it?

Comment (anandtiwary) [reply projects/components/src/popover/popover.service.ts:89]: I don't think there is any benefit. It sets the opacity. I think we can leave it as it is till we modify the style.

**Pull request merged**